rest/authentication.go

```diff
@@ -0,0 +1,106 @@
+package rest
+
+import (
+	"errors"
+	"github.com/aueb-cslabs/moniteur/types"
+	"github.com/dgrijalva/jwt-go"
+	"github.com/labstack/echo"
+	"net/http"
+	"strings"
+	"time"
+)
+
+func Authenticate(e echo.Context) error {
+	ctx := e.(*types.Context)
+	user := &types.User{}
+
+	err := e.Bind(user)
+
+	if err != nil {
+		return e.JSON(http.StatusBadRequest, err)
+	}
+
+	res, ldapErr := ctx.Plugin().AuthorizeUser(user.Username, user.Password)
+
+	if res {
+		expiresAt := time.Now().Add(time.Hour * 24).Unix()
+		token := jwt.New(jwt.SigningMethodHS256)
+
+		standardClaim := &jwt.StandardClaims{ExpiresAt: expiresAt}
+
+		authTokenClaim := &types.AuthTokenClaim{}
+		authTokenClaim.StandardClaims = standardClaim
+		authTokenClaim.Username = user.Username
+
+		token.Claims = authTokenClaim
+
+		tokenString, err := token.SignedString([]byte(secret))
+		if err != nil {
+			return e.JSON(http.StatusUnauthorized, err)
+		}
+
+		authToken := &types.AuthToken{}
+		authToken.Token = tokenString
+		authToken.TokenType = "Bearer"
+		authToken.ExpiresIn = expiresAt
+
+		authorized[authToken.Token] = authTokenClaim
+
+		e.Response().Header().Set("Authorization", "Bearer "+authToken.Token)
```

Comment:
Do not write these as response headers, pass them into a JSON object and let the frontend decide on what to do with them.

Reply:
Πλέον επιστρέφει όλο το auth token.